src/m365/spo/commands/term/term-get.ts

```diff
@@ -132,49 +126,77 @@ class SpoTermGetCommand extends SpoCommand {
       }
 
-      const requestOptions: any = {
-        url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
-        headers: {
-          'X-RequestDigest': res.FormDigestValue
-        },
-        data: data
-      };
-
-      const processQuery: string = await request.post(requestOptions);
-      const json: ClientSvcResponse = JSON.parse(processQuery);
-      const response: ClientSvcResponseContents = json[0];
-      if (response.ErrorInfo) {
-        throw response.ErrorInfo.ErrorMessage;
+      let term: Term;
+      const csomResponse = await this.executeCsomCall(data, spoAdminUrl, res);
+
+      if (csomResponse === null) {
+        throw `Term with id '${args.options.id}' could not be found.`;
       }
+      else if (csomResponse._Child_Items_ !== undefined) {
+        const terms: TermCollection = csomResponse;
+        if (terms._Child_Items_.length === 0) {
+          throw `Term with name '${args.options.name}' could not be found.`;
+        }
+        if (terms._Child_Items_.length > 1) {
+          const disambiguationText = terms._Child_Items_.map(c => {
+            return `- ${this.replaceTermId(c.Id)} - ${c.PathOfTerm}`;
+          }).join(os.EOL);
 
-      const term: Term | null = json[json.length - 1];
-      if (!term) {
-        return;
+          throw new Error(`Multiple terms with the specific term name found. Please disambiguate:${os.EOL}${disambiguationText}`);
+        }
+        term = terms._Child_Items_[0];
+      }
+      else {
+        term = csomResponse;
       }
 
       delete term._ObjectIdentity_;
       delete term._ObjectType_;
-      term.CreatedDate = new Date(Number(term.CreatedDate.replace('/Date(', '').replace(')/', ''))).toISOString();
-      term.Id = term.Id.replace('/Guid(', '').replace(')/', '');
-      term.LastModifiedDate = new Date(Number(term.LastModifiedDate.replace('/Date(', '').replace(')/', ''))).toISOString();
+      term.CreatedDate = this.parseTermDateToIsoString(term.CreatedDate);
+      term.Id = this.replaceTermId(term.Id);
+      term.LastModifiedDate = this.parseTermDateToIsoString(term.LastModifiedDate);
       logger.log(term);
-
     }
     catch (err: any) {
       this.handleRejectedPromise(err);
     }
   }
+
+  private async executeCsomCall(data: string, spoAdminUrl: string, res: ContextInfo): Promise<any> {
+    const requestOptions: CliRequestOptions = {
+      url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+      headers: {
+        'X-RequestDigest': res.FormDigestValue
+      },
+      data: data
+    };
+
+    const processQuery: string = await request.post(requestOptions);
+    const json: ClientSvcResponse = JSON.parse(processQuery);
+    const response: ClientSvcResponseContents = json[0];
+    if (response.ErrorInfo) {
+      throw response.ErrorInfo.ErrorMessage;
+    }
+    const responseObject = json[json.length - 1];
+    return responseObject;
+  }
+
+  private replaceTermId(termId: string): string {
```

Comment:
since the aim of this method is to get the id of the term (or like 'clean' the output to get the id by using replace) I think better naming for this would be `getTermId`. 🤔 
Wouldn't you agree ?

Reply:
Makes sense indeed.

Reply:
great thanks.